### PR TITLE
Add ReviewEditor component

### LINKED ISSUE #31: Review writing popup

### PR DESCRIPTION
**Pull request by Qmpzlawasd**
Closes #31

### CODE REVIEW

File: frontend/src/pages/books/[id].tsx [hunk 1]
```diff
@@ -4,32 +4,121 @@ import { css } from "@emotion/react"
...
 
-const mockBook = { id: 1, title: "Book1", author: "Author1", rating: 3, coverImageUrl: 'https://picsum.photos/300/480' }
-const reviews = [{ id: 1, rating: 4, text: "Pretty good book", authorUsername: 'the_critic' }, { id: 2, rating: 2, text: "GarBAGE", authorUsername: 'hater' }]
+interface StarsProps {
+  ratingMax: number;
+  stars: number;
+  setStars: Function;
+}
+
+function Stars({ ratingMax, stars, setStars }: StarsProps) {
```
Comment (tezkerek) [inline frontend/src/pages/books/[id].tsx:17]: This can be an independent component in its own file.

File: frontend/src/pages/books/[id].tsx [hunk 2]
```diff
@@ -4,32 +4,121 @@ import { css } from "@emotion/react"
...
+import TextareaAutosize from "@mui/material/TextareaAutosize"
+import Button from "@mui/material/Button"
+import { useState } from "react"
 
-const mockBook = { id: 1, title: "Book1", author: "Author1", rating: 3, coverImageUrl: 'https://picsum.photos/300/480' }
-const reviews = [{ id: 1, rating: 4, text: "Pretty good book", authorUsername: 'the_critic' }, { id: 2, rating: 2, text: "GarBAGE", authorUsername: 'hater' }]
+interface StarsProps {
+  ratingMax: number;
+  stars: number;
+  setStars: Function;
```
Comment (tezkerek) [inline frontend/src/pages/books/[id].tsx:14]: Naming: `maxRating`, `rating`, `onRatingChange`

Comment (tezkerek) [inline frontend/src/pages/books/[id].tsx:14]: Add an actual signature to `onRatingChange`, like `(rating: number) => void`.

File: frontend/src/pages/books/[id].tsx (hunk 1 above)
Comment (tezkerek) [inline frontend/src/pages/books/[id].tsx:17]: Naming: `RatingSelector`?

File: frontend/src/pages/books/[id].tsx [hunk 3]
```diff
@@ -4,32 +4,121 @@ import { css } from "@emotion/react"
...
+          ></path>
+        </svg>
+      </div>
+    );
+  }
+
+  return <div>{starArr}</div>;
+}
+
+function WriteReviewPopup(): JSX.Element {
```
Comment (tezkerek) [inline frontend/src/pages/books/[id].tsx:44]: Naming: `ReviewEditor`? We'll probably use it to edit reviews as well.

File: frontend/src/pages/books/[id].tsx [hunk 4]
```diff
@@ -4,32 +4,121 @@ import { css } from "@emotion/react"
...
+          flex-direction: column;
+          justify-content: space-between;
+          align-items: center;
+        `}
+      >
+        <p>Write review</p>
+
+        <Stars ratingMax={5} stars={stars} setStars={setStars} />
+
+        <TextareaAutosize
```
Comment (tezkerek) [inline frontend/src/pages/books/[id].tsx:62]: Make the initial textarea larger.

File: frontend/src/pages/books/[id].tsx [hunk 5]
```diff
@@ -4,32 +4,121 @@ import { css } from "@emotion/react"
...
+        </svg>
+      </div>
+    );
+  }
+
+  return <div>{starArr}</div>;
+}
+
+function WriteReviewPopup(): JSX.Element {
+  const [textReview, setTextReview] = useState<string>("");
```
Comment (tezkerek) [inline frontend/src/pages/books/[id].tsx:45]: Naming is backwards: this is the review's text, not the text's review.

File: frontend/src/pages/books/[id].tsx [hunk 6]
```diff
@@ -4,32 +4,121 @@ import { css } from "@emotion/react"
...
+  const [stars, setStars] = useState<number>(0);
```
Comment (tezkerek) [inline frontend/src/pages/books/[id].tsx:46]: Naming: use `rating` for consistency with the Review entity.

File: frontend/src/pages/books/[id].tsx [hunk 7]
```diff
@@ -4,32 +4,121 @@ import { css } from "@emotion/react"
...
+          onChange={(e: React.ChangeEvent<HTMLTextAreaElement>) => {
+            setTextReview(e.target.value);
+          }}
+        />
+        <Button
+          color="warning"
+          variant="outlined"
+          disabled={!(stars >= 1 && stars <= 5)}
+          onClick={() => {
+            console.log(textReview, stars);
```
Comment (tezkerek) [inline frontend/src/pages/books/[id].tsx:76]: If you add a `onDone: (review: Review) => void` prop, you can make this an independent component too.

File: frontend/src/pages/books/[id].tsx [hunk 8]
```diff
@@ -4,32 +4,121 @@ import { css } from "@emotion/react"
 import BookDetail from "@/components/BookDetail"
 import Navbar from "@/components/Navbar"
 import ReviewList from "@/components/ReviewList"
+import TextareaAutosize from "@mui/material/TextareaAutosize"
+import Button from "@mui/material/Button"
+import { useState } from "react"
 
-const mockBook = { id: 1, title: "Book1", author: "Author1", rating: 3, coverImageUrl: 'https://picsum.photos/300/480' }
-const reviews = [{ id: 1, rating: 4, text: "Pretty good book", authorUsername: 'the_critic' }, { id: 2, rating: 2, text: "GarBAGE", authorUsername: 'hater' }]
```
Comment (tezkerek) [inline frontend/src/pages/books/[id].tsx:9]: Don't change unrelated lines.

File: frontend/src/pages/books/[id].tsx [hunk 9]
```diff
@@ -4,32 +4,121 @@ import { css } from "@emotion/react"
...
+          }}
+        >
+          Post review
+        </Button>
+      </div>
+    </>
+  );
+}
+
+const mockBook = {
```
Comment (tezkerek) [inline frontend/src/pages/books/[id].tsx:86]: Here and below: don't change unrelated lines.

**Review by tezkerek — CHANGES_REQUESTED** [review]

File: frontend/src/pages/books/[id].tsx (hunk 7 above)
Comment (Qmpzlawasd) [reply frontend/src/pages/books/[id].tsx]: > If you add a `onDone: (review: Review) => void` prop, you can make this an independent component too.
Could you be more specific?

Comment (tezkerek) [reply frontend/src/pages/books/[id].tsx]: When it's finished editing, `ReviewEditor` should return the updated/created review to its parent via the `onDone` callback. And now `ReviewEditor` should be in its own file.

**Review by tezkerek — APPROVED** [review]

**Pull request merged**